java/compaction/compaction-core/src/test/java/sleeper/compaction/strategy/impl/BasicCompactionStrategyTest.java

```diff
@@ -67,71 +68,69 @@ public void shouldCreateOneJobWhenOneLeafPartitionAndOnlyTwoFiles() {
         tableProperties.set(COMPACTION_FILES_BATCH_SIZE, "2");
         BasicCompactionStrategy strategy = new BasicCompactionStrategy();
-        strategy.init(instanceProperties, tableProperties);
         PartitionTree partitionTree = new PartitionsBuilder(DEFAULT_SCHEMA)
                 .singlePartition("root")
                 .buildTree();
         FileReferenceFactory factory = FileReferenceFactory.from(partitionTree);
         FileReference fileReference1 = factory.rootFile("file1", 100L);
         FileReference fileReference2 = factory.rootFile("file2", 100L);
         List<FileReference> fileReferences = List.of(fileReference1, fileReference2);
+        CompactionJobFactory jobFactory = fixJobIds(List.of("job1"));
 
         // When
-        List<CompactionJob> compactionJobs = strategy.createCompactionJobs(List.of(), fileReferences, partitionTree.getAllPartitions());
+        List<CompactionJob> compactionJobs = strategy.createCompactionJobs(
+                instanceProperties, tableProperties, jobFactory, fileReferences, partitionTree.getAllPartitions());
 
         // Then
-        assertThat(compactionJobs).hasSize(1);
-        CompactionJob expectedCompactionJob = jobForTable()
-                .jobId(compactionJobs.get(0).getId()) // Job id is a UUID so we don't know what it will be
-                .partitionId("root")
-                .inputFiles(List.of("file1", "file2"))
-                .outputFile("file://databucket/table-id/data/partition_root/" + compactionJobs.get(0).getId() + ".parquet")
-                .iteratorClassName(null)
-                .iteratorConfig(null).build();
-        assertThat(compactionJobs).containsExactly(expectedCompactionJob);
+        assertThat(compactionJobs).containsExactly(
+                jobFactory.createCompactionJob("job1", fileReferences, "root"));
     }
 
     @Test
     public void shouldCreateCorrectJobsWhenOneLeafPartitionAndLotsOfFiles() {
         // Given
         tableProperties.set(COMPACTION_FILES_BATCH_SIZE, "10");
         BasicCompactionStrategy strategy = new BasicCompactionStrategy();
-        strategy.init(instanceProperties, tableProperties);
         PartitionTree partitionTree = new PartitionsBuilder(DEFAULT_SCHEMA)
                 .singlePartition("root")
                 .buildTree();
         FileReferenceFactory factory = FileReferenceFactory.from(partitionTree);
         List<FileReference> fileReferences = new ArrayList<>();
+        List<FileReference> filesInAscendingOrder = new ArrayList<>();
         for (int i = 0; i < 100; i++) {
             FileReference fileReference = factory.rootFile("file-" + i, 1_000_000L - i * 100L);
             fileReferences.add(fileReference);
+            filesInAscendingOrder.add(0, fileReference);
         }
+        CompactionJobFactory jobFactory = fixJobIds(
+                IntStream.iterate(1, i -> i + 1)
+                        .mapToObj(i -> "job" + i)
+                        .iterator()::next);
```

Comment:
It might be useful to put this in a helper method to generate autoincrementing job IDs?